Gen 8 Update for the upcoming event

Audidriver13 opened the pull request:
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->
<!--         PULL REQUESTS CREATED NOT USING THE TEMPLATE           --->
<!--              WILL BE CLOSED WITHOUT ANY RESPONSE               --->


<!--
 Please make all PRs to the dev branch. The dev branch will be
 periodically pulled into the master branch. This allows time for
 the changes and documentation to be tested before being exposed to
 a wider population.
--->

## Description
<!-- In detail, describe what your PR adds to PokeAlarm -->
Part 3 of the hyperbonus event starts on August 20th with the new Generation 8 Pokémon. In preparation for this, all values of Gen8 were added, the PVP stats were recalculated (base_stats.json) and the Pokémon names were added in the locales (de, en, fr). The (unnecessary?) Information on the height and weight of the Pokémon was omitted due to time constraints.

Note: The calculation for generation 7 is still incorrect because no correct base values are available yet.

## Type of Change
<!-- Place a single 'x' into the correct box, ex: [x] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Wiki Update
<!--
 Does this feature require an update to the wiki? If so, please submit
 the required change to https://github.com/RocketMap/PokeAlarmWiki.
 If your feature requires a wiki update, you may submit it for review
 but it will not be accepted until the wiki update is complete.
--->
- [ ] This change requires an update to the Wiki.
- [x] This change does not require an update to the Wiki.

Audidriver13 committed:
Add files via upload

Audidriver13 committed:
Add files via upload

Audidriver13 committed:
Add files via upload

Audidriver13 commented:
If this PR is accepted, PR # 771 can be closed because it has been implemented here.

TiMXL73 commented:
Gen 8 is going to be live in less than 6h in New Zealand and PA still doesn't support Gen 8 until this or an updated variation of it is merged.

NOTE: The updated locale json are using TABS instead of SPACES for their indentation. This should eventually be fixed along with adding dummy/temporary values for the other locale languages to avoid errors (ie, use english for now and fix later?).

Audidriver13 committed:
Add files via upload

replace TABs with SPACES and create english dummys for the other languages (es,it,ko,pt,zh_hk)

Audidriver13 commented:
@TiMXL73 have implemented your notes :) thanks for the hint

LegitDongo commented on locales/en.json:
```diff
@@ -4590,7 +4590,7 @@
     "770": "Palossand",
     "771": "Pyukumuku",
-    "772": "Type: Null",
+    "772": "TypeNull",
```
This, along with the Tapu mons, doesn't look correct, is this how it is in-game?

Audidriver13 replied:
The correct name is "Type: Null" (https://www.pokemon.com/us/pokedex/type-null), but I changed it to "TypeNull" to be on the safe side so that there are no errors.

Audidriver13 replied:
Oh. The Tapu Mons is a typical case of "finding and replacing spaces". I'll put the space right back in

LegitDongo replied:
Special characters and spaces aren't a problem for PA, the main problem there lies when the encoding is outside of UTF-8, because of how the json files themselves are encoded
This is supplemented elsewhere by giving python unicode strings (the "\u..." you see here and there) to represent the properly encoded string representation

Back on topic, references to this are usually done in a string, such as in filters: `monsters: ["Type: Null"]`, so spaces aren't a problem

Audidriver13 replied:
Tapu mons changed with spaces. The Type: Null mon ... you can decide, should I use the original name (Type: Null) or leave it that way (TypeNull)?

LegitDongo replied:
change it to `Type: Null` please 😄

Audidriver13 replied:
done :)

LegitDongo reviewed (changes requested):
Just one complaint

Thanks for the amendments, I had planned on making them myself but have found my time lacking of late

Audidriver13 committed:
Add files via upload

add spaces on "Tapu" mons

Audidriver13 committed:
Add files via upload

changed TypeNull mon to Type: Null

LegitDongo reviewed (approved):
This will do for now, will go back over this some other time
